### PR TITLE
Add websockets support

### PR DESCRIPTION
**Pull request by ybart**
Rebased https://github.com/code-mancers/invoker/pull/156 on master

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,9 @@ source "https://rubygems.org"
 
 gemspec
 
+gem 'websocket-eventmachine-server', require: false
+gem 'websocket-eventmachine-client', require: false
+
```
Comment (gnufied) [inline Gemfile]: Why is this here but not in gemspec? Does this mean, we actually don't need invoker to understand websocket to process the requests?

**Comment by iffyuva** [comment]
this is duplicate of https://github.com/code-mancers/invoker/pull/156

File: Gemfile (hunk 1 above)
Comment (iffyuva) [inline Gemfile]: they are test dependencies. no need to be part of gemspec i suppose.

invoker doesn't have to understand websockets. if server keeps tcp connection open, invoker is transparently keeping that connection open. this behavior saved lots of coding.

**Comment by ybart** [comment]
@iffyuva Yes, I've simply rebased #156 on master.

**Comment by gnufied** [comment]
I am thinking of merging this @iffyuva unless you have an objection,

**Comment by iffyuva** [comment]
no issues, please go ahead and merge it. thanks.

On Wed, Jan 20, 2016 at 6:13 AM, Hemant Kumar <notifications@github.com>
wrote:

> I am thinking of merging this @iffyuva <https://github.com/iffyuva>
> unless you have an objection,
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-mancers/invoker/pull/158#issuecomment-173035516>.
>



-- 
Yuva (@iffyuva)
Co-founder at Codemancers Tech Pvt Ltd
http://codemancers.com
